### PR TITLE
copr: Add vectorized json-length

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Wangweizhen <hawking.rei@gmail.com>

PCP #5751

###  What have you changed?

Add vectorized json-length

###  What is the type of the changes?

- New feature

###  How is the PR tested?

Unit tested. Will run copr test later.

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No. It should not change the behavior.

###  Does this PR affect `tidb-ansible`?

No.

### CODE REVIEW

**Comment by hawkingrei** [comment]
@sre-bot  /run-integration-copr-test copr-test

File: components/tidb_query/src/rpn_expr/impl_json.rs [hunk 1]
```diff
@@ -187,6 +187,34 @@ fn json_extract(args: &[ScalarValueRef]) -> Result<Option<Json>> {
     Ok(j.extract(&path_expr_list))
 }
 
+// Args should be like `(&Option<Json> , &[&Option<Bytes>])`.
+fn json_with_path_validator(expr: &tipb::Expr) -> Result<()> {
+    let children = expr.get_children();
+    assert!(children.len() == 1 || children.len() == 2);
```
Comment (sticnarf) [inline components/tidb_query/src/rpn_expr/impl_json.rs]: I think an error should returned instead. Invalid request from TiDB shouldn't make TiKV panic.

Comment (hawkingrei) [reply components/tidb_query/src/rpn_expr/impl_json.rs]: I agree with you. But I find  that ```json_with_paths_validator``` contains ```assert!``` too. so I will fix it and open new PR to fix ```json_with_paths_validator```

**Comment by hawkingrei** [comment]
@sticnarf PTAL

File: components/tidb_query/src/rpn_expr/impl_json.rs [hunk 2]
```diff
@@ -187,6 +187,38 @@ fn json_extract(args: &[ScalarValueRef]) -> Result<Option<Json>> {
     Ok(j.extract(&path_expr_list))
 }
 
+// Args should be like `(&Option<Json> , &[&Option<Bytes>])`.
+fn json_with_path_validator(expr: &tipb::Expr) -> Result<()> {
```
Comment (breeswish) [inline components/tidb_query/src/rpn_expr/impl_json.rs]: Can you merge the implementation with `json_with_paths_validator`?

File: components/tidb_query/src/rpn_expr/impl_json.rs [hunk 3]
```diff
@@ -187,6 +191,27 @@ fn json_extract(args: &[ScalarValueRef]) -> Result<Option<Json>> {
...
+    assert!(expr.get_children().len() == 2 || expr.get_children().len() == 1);
+    valid_paths(expr)
+}
+
+#[rpn_fn(raw_varg, min_args = 2, extra_validator = json_with_path_validator)]
+#[inline]
+fn json_length(args: &[ScalarValueRef]) -> Result<Option<Int>> {
```
Comment (breeswish) [inline components/tidb_query/src/rpn_expr/impl_json.rs]: Maybe you need to add `max_args` to avoid TiKV panic when wrong number of arguments is passed?

Comment (hawkingrei) [reply components/tidb_query/src/rpn_expr/impl_json.rs]: Yes，I can add max_args to rpn_fn. so I need to open new pr to implement this.

Comment (hawkingrei) [reply components/tidb_query/src/rpn_expr/impl_json.rs]: Done

File: components/tidb_query/src/rpn_expr/impl_json.rs (hunk 2 above)
Comment (hawkingrei) [reply components/tidb_query/src/rpn_expr/impl_json.rs]: Done

**Comment by hawkingrei** [comment]
@breeswish PTAL

**Comment by hawkingrei** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/212

**Comment by breeswish** [comment]
@sticnarf @TennyZhuang PTAL thanks!

**Comment by TennyZhuang** [comment]
LGTM

**Pull request merged**